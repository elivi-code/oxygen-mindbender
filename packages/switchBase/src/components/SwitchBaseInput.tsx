import React, { ChangeEventHandler } from 'react';

import { getTestAttributes } from '@8x8/oxygen-config';

import { SwitchBaseInputField, SwitchBaseInputWrapper } from '../styled';
import { SwitchBaseCommonProps } from './SwitchBaseCommonProps';
import { InputValue } from '../types';

export interface Props<T extends InputValue> extends SwitchBaseCommonProps<T> {
  onChange?: ChangeEventHandler<HTMLInputElement>;
  onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
}

const SwitchBaseInput = <T extends InputValue>({
  testId,
  id,
  value,
  name,
  icon,
  isChecked,
  isDisabled,
  onBlur,
  onChange,
  onFocus,
  onMouseEnter,
  onMouseLeave,
  onKeyPress,
  role,
  ...rest
}: Props<T>) => {
  return (
    <SwitchBaseInputWrapper>
      <SwitchBaseInputField
        {...getTestAttributes(testId as string)}
        id={id}
        value={value}
        type="checkbox"
        name={name}
        checked={isChecked}
        aria-disabled={isDisabled}
        onBlur={onBlur}
        onChange={onChange}
        onFocus={onFocus}
        onKeyPress={onKeyPress}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
        role={role}
        {...rest}
      />
      {icon}
    </SwitchBaseInputWrapper>
  );
};

export default SwitchBaseInput;
