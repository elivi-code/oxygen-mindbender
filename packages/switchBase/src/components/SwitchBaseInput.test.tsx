import React from 'react';
import renderer from 'react-test-renderer';

import SwitchBaseInput, { Props } from './SwitchBaseInput';

jest.mock('../styled', () => ({
  __esModule: true,
  SwitchBaseInputField: 'SwitchBaseInputField',
  SwitchBaseInputWrapper: 'SwitchBaseInputWrapper',
}));

describe('<SwitchBaseInput />', () => {
  let wrapper;
  const setup = (props?: Omit<Props, 'id' | 'icon' | 'testId' | 'onChange'>) =>
    renderer
      .create(
        <SwitchBaseInput
          testId="testId"
          name="test-name"
          id="test-id"
          isChecked={false}
          isDisabled={false}
          onBlur={jest.fn()}
          onChange={jest.fn()}
          onFocus={jest.fn()}
          onMouseEnter={jest.fn()}
          onMouseLeave={jest.fn()}
          icon={null}
          {...props}
        />,
      )
      .toJSON();

  it('should render correctly with default props', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with role', () => {
    wrapper = setup({ role: 'testRole' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly is disabled', () => {
    wrapper = setup({ isDisabled: true });
    expect(wrapper).toMatchSnapshot();
  });
});
