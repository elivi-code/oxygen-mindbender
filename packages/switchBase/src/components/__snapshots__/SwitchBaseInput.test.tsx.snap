// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwitchBaseInput /> should render correctly is disabled 1`] = `
<SwitchBaseInputWrapper>
  <SwitchBaseInputField
    aria-disabled={true}
    checked={false}
    data-id="testId"
    data-test-id="testId"
    id="test-id"
    name="test-name"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    onMouseEnter={[MockFunction]}
    onMouseLeave={[MockFunction]}
    test-id="testId"
    type="checkbox"
  />
</SwitchBaseInputWrapper>
`;

exports[`<SwitchBaseInput /> should render correctly with default props 1`] = `
<SwitchBaseInputWrapper>
  <SwitchBaseInputField
    aria-disabled={false}
    checked={false}
    data-id="testId"
    data-test-id="testId"
    id="test-id"
    name="test-name"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    onMouseEnter={[MockFunction]}
    onMouseLeave={[MockFunction]}
    test-id="testId"
    type="checkbox"
  />
</SwitchBaseInputWrapper>
`;

exports[`<SwitchBaseInput /> should render correctly with role 1`] = `
<SwitchBaseInputWrapper>
  <SwitchBaseInputField
    aria-disabled={false}
    checked={false}
    data-id="testId"
    data-test-id="testId"
    id="test-id"
    name="test-name"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    onMouseEnter={[MockFunction]}
    onMouseLeave={[MockFunction]}
    role="testRole"
    test-id="testId"
    type="checkbox"
  />
</SwitchBaseInputWrapper>
`;
